core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -1299,11 +1299,7 @@ public Builder removeRef(String name) {
 
     private Builder resetMainBranch() {
       this.currentSnapshotId = -1;
-      SnapshotRef ref = refs.remove(SnapshotRef.MAIN_BRANCH);
-      if (ref != null) {
-        changes.add(new MetadataUpdate.RemoveSnapshotRef(SnapshotRef.MAIN_BRANCH));
-      }
-
+      setBranchSnapshot(-1, SnapshotRef.MAIN_BRANCH);
```

Comment:
Leaving this in draft, there's a good argument that we should update the removeRef(Main) behavior to not do the snapshotlog.clear() but in the past I was worried about behavior which relied on that.